release notes/v0.45.0.md

```diff
@@ -0,0 +1,169 @@
+
+  stream.on('data', function (feature) {
+    console.log(
+      'Found feature called "' +
+        feature.name +
+        '" at ' +
+        feature.location.latitude / COORD_FACTOR +
+        ', ' +
+        feature.location.longitude / COORD_FACTOR
+    );
+  });
+
+  stream.on('end', function () {
+    // The server has finished sending
+    client.close();
+    console.log('All done');
+  });
+
+  stream.on('error', function (e) {
+    // An error has occurred and the stream has been closed.
+    console.log('Error: ' + JSON.stringify(e));
+  });
+
+  // send a message to the server
+  stream.write({
+    lo: {
+      latitude: 400000000,
+      longitude: -750000000,
+    },
+    hi: {
+      latitude: 420000000,
+      longitude: -730000000,
+    },
+  });
+
+  sleep(0.5);
+};
+```
+</details>
+
+You can just replace `k6/grpc` import with `k6/experimental/grpc` to use the new functionality.
+
+In the future this functionality will be moved to the `k6/grpc` module.
+
+### You can now only upload a test to the cloud without running it [#3030](https://github.com/grafana/k6/pull/3030)
+
+For [years users have wanted](https://github.com/grafana/k6-cloud-feature-requests/issues/22) to be able to update the test that is saved in the cloud but *not* run it at this exact point.
+
+This is now possible by adding `--upload-only` when invoking `k6 cloud` as in `k6 cloud --upload-only script.js`.
+
+This is likely going to be most useful in a CI on the actual test script project. Now that CI can just run `k6 cloud --upload-only new-verson-of-script.js` on "release".
+
+And later on that newer version will be used. For example by a [scheduled run](https://k6.io/docs/cloud/manage/scheduled-tests/). 
+
+### Setting sample metadata API [#3037](https://github.com/grafana/k6/pull/3037)
+
+It is now possible to set metadata for samples emitted by a given VU. 
+
+Metadata was added in v0.41.0 but was not available to be set from JS code. This was mainly due to lack of agreement on the API to do so.
+
+After further discussion, a compromised, smaller and consistent API was agreed upon.
+
+The API is similar to the one used to set tags for all metrics of the VU:
+```javascript
+import exec from "k6/execution";
+
+export default () => {
+    exec.vu.metrics.metadata["my_cool_id"] = "a very unique value"
+    // all metrics from here on will have this metadata set
+    delete exec.vu.metrics.metadata["my_cool_id"]
+    // all metriccs from here on wil *not* have the metadata set
+}
+```
+
+This also introduces the sub-object `metrics` on the `vu` object.
+Apart from `metadata` it has another property `tags`. This is supposed to be the new way to set tags instead of usng `exec.vu.tags`. 
+
+There are no current plans when or if `exec.vu.tags` will be dropped and only `exec.vu.metrics.tags` to stay.
+
+### UX improvements and enhancements
+
+- [#3015](https://github.com/grafana/k6/pull/3015) Official arm64 docker images. Thanks to @nrxr! 🙇
```

Comment:
There is quite a lot of writing most of which is some technical details. 

But if there is no benefit from the changes shall we revert? If there are benefits we can mention them

Reply:
There are no practical benefits for users. We build the ARM image in CI, but it's not pushed anywhere.

We don't need to revert it, since there's no harm in leaving it as is, but I'm hoping that the author can address the push issue for the next release, and we can mention this then.

Reply:
```suggestion
- [#3015](https://github.com/grafana/k6/pull/3015) We started building ARM64 Docker images. There is still not pushed in the official repo though. Thanks to @nrxr! 🙇
```

Reply:
I don't think we should mention it at all... There's literally no way for people to benefit from this, and it will only raise questions about how to use it. I would like to credit the author for the work done so far as well, but it doesn't make sense until the work is actually finished.

Reply:
On second thought, it probably deserves to be mentioned. To credit the author, and to point out that we're working on it, even though it's not usable yet. So maybe:

```suggestion
- [#3015](https://github.com/grafana/k6/pull/3015) We're now building ARM64 Docker images in CI, but due to some Docker complications, they're still not pushed to the repository. We expect to resolve these issues by next release. Thanks @nrxr! 🙇
```